rust_src/src/buffers.rs

```diff
@@ -562,6 +620,84 @@ pub fn overlay_lists() -> LispObject {
     unsafe { LispObject::from_raw(Fcons(Fnreverse(before.to_raw()), Fnreverse(after.to_raw()))) }
 }
 
+fn get_truename_buffer_1(filename: LispObject) -> LispObject {
+    LispObject::from_raw(unsafe { Vbuffer_alist })
+        .iter_alist_vals()
+        .find(|buf| {
+            let buf_truename = buf.as_buffer_or_error().truename();
+            buf_truename.is_string() && string_equal(buf_truename, filename)
+        })
+        .and_then(|obj| obj.as_buffer())
+        .into()
+}
```

Comment:
Would it make send to add `.map(|buf| buf.as_buffer_or_error())` before the `find`? This would make it so we only do the conversion once. If we do this a bunch it may even make sense to emulate the `FOR_EACH_LIVE_BUFFER` and have an explicit `Buffer` iterator.

Reply:
well, I would prefer a generic `AlistValIter<T>` so that we can use `type BufferIter = AlistValIter<LispBufferRef>`.

Reply:
That sounds reasonable.